server/server.js

```diff
@@ -113,198 +120,209 @@ function buildRegExp (searchText) {
+      });
+    },
+    /**
+     * @summary Add listing with params
+     * @locus Anywhere
+     * @instancename collection
+     * @class
+     */
+    addListing: function(options) {
+      if (!Meteor.userId()) {
+        throw new Meteor.Error("Not Authorized")
+      }
+
+      if (options.trade == "1") {
+        options.trade = "Trades Allowed";
+      } else {
+        options.trade = "Trades Denied";
+      }
+
+      Listing.insert({
+        createdAt: new Date(),
+        creator_id: options.creator_id,
+        facebook_id: options.facebook_id,
+        listing_title: options.listing_title,
+        category: options.category,
+        type: options.type,
+        brand: options.brand,
+        username: Meteor.user().profile.name,
+        quantity: options.quantity,
+        price: options.price,
+        // city: options.city,
+        // state: options.state,
+        trade: options.trade,
+        size: options.size,
+        condition: options.condition,
+        color: options.color,
+        description: options.description,
+        lat: options.lat,
+        lng: options.lng,
+        img1: options.img1,
+        img2: options.img2,
+        img3: options.img3
+      });
+    },
+    searchFilter: function(option) {
+      // console.log(options.Categories); 
+
+      //   var selectorCat = {
+      //   category: options.Categories
+      // }
+
+      var selectorCity = {
+        city: 'Queens'
+      };
+
+      var selectorCategory = {
+        category: 'Apparel'
+      };
+
+      console.log(selectorCategory);
+
+      var query = Listing.find({
+        $and: [selectorCategory, selectorCity]
+      }).fetch();
+      console.log(query);
+      return query;
+    },
+    addOffer: function(options) {
+      Offer.insert({
+        listing_title: options.listing_title,
+        img: options.img,
+        offerprice: options.offerprice,
+        date: options.date,
+        location: options.location,
+        listingId: options.listingId,
+        lat: options.lat,
+        lng: options.lng,
+        creator_id: options.creator_id,
+        status: options.status
+      });
+    },
+    addProfileInfo: function(options) {
+      Meteor.users.update(this.userId, {
+        $set: {
+          'profile.photo': options.photo,
+          'profile.about': options.about,
+          'profile.email': options.email,
+          'profile.location': options.location,
+          'profile.link': options.link
+        }
+});  },
+  // IDEAL: instead of manually typing in keys, use iteration
+  addReport: function(options) {
+    Report.insert({
+        targetUser: options.targetUser,
+        riskLevel: options.riskLevel,
+        reasonBox: {
+          prohibited_box: options.reasonBox.prohibited_box,
+          offensive_box: options.reasonBox.offensive_box,
+          irrelevant_box: options.reasonBox.irrelevant_box,
+          false_box: options.reasonBox.false_box,
+          compliance_box: options.reasonBox.compliance_box
```

Comment:
Identifier 'compliance_box' is not in camel case.<br>Too many errors. (82% scanned).